test/freenet/client/filter/ContentFilterTest.java

```diff
@@ -380,6 +380,37 @@ public void testEvilCharset() throws IOException {
         }
     }
 
+
+    @Test
+    public void byteOrderMarkForUtf8IsDetectedCorrectly() throws IOException {
+        byte[] buf = { (byte) 0xef, (byte) 0xbb, (byte) 0xbf, 0x40 };
+        ArrayBucket out = new ArrayBucket();
+        FilterStatus fo = ContentFilter.filter(new ArrayBucket(buf).getInputStream(), out.getOutputStream(), "text/plain", null, null, null);
+        assertTrue("utf-8".equals(fo.charset));
```

Comment:
Please use `assertThat(actual, equalTo(expected))` or `assertEquals(expected, actual)` for checking equality - this just yields a non-descriptive AssertionError when it fails instead of showing what the actual value was.